samples/Rewrite/FileParser/RewriteConfigurationFileParser.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Rewrite.Structure2;
+using Rewrite.ConditionParser;
+using Rewrite.RuleParser;
+namespace Rewrite.FileParser
+{
+    public static class RewriteConfigurationFileParser
+    {
+        private static StreamReader _reader;
+        public static List<Rule> Parse(Stream input)
+        {
+            _reader = new StreamReader(input);
+            var line = (string) null;
+            List<Rule> rules = new List<Rule>();
+            var currentRule = new Rule { Conditions = new List<Condition>() };
+            while ((line = _reader.ReadLine()) != null) {
+                // TODO handle comments
+                List<string> tokens = RewriteTokenizer.TokenizeRule(line);
+                if (tokens.Count < 3)
+                {
+                    // This means the line didn't have an appropriate format, throw format exception
+                    throw new FormatException();
+                }
+                switch (tokens[0])
+                {
+                    case "RewriteCond":
```

Comment:
The file parser shouldn't know anything about how a rule looks like. Delegate this to the rule parser. Make the file parser only call into the rule parser and get rules or get an exception back